pkg/internal/goexec/structmembers.go

```diff
@@ -115,8 +115,12 @@ func structMemberOffsets(elfFile *elf.File) (FieldOffsets, error) {
 	var expected map[string]struct{}
 	dwarfData, err := elfFile.DWARF()
 	if err == nil {
-		offs, expected = structMemberOffsetsFromDwarf(dwarfData)
+		_, _, err := getGoDetails(elfFile)
```

Comment:
Sorry, why is this needed?

Reply:
I believe we need this because the library versions are only checked if we can't find the symbols, in `structMemberPreFetchedOffsets`. If the executable has symbols and it's compiled with older Go version we won't check otherwise.

Reply:
This suggestion came from the discussion in [the related PR][1]:

> Finally, I think we should also call the `getGoDetails()` function at this line here:
>
> https://github.com/grafana/beyla/blob/760d3ad8b21dc880606b0d5d0d60f93c1ed02ba7/pkg/internal/goexec/structmembers.go#L123
>
> In case we find go symbols in the ELF we wouldn't bother checking for what's the Go version, we trust what the ELF says the offsets are. A simple call with only caring about the returned err should suffice. If we get an error there we can just propagate it upward to make the go instrumentation fail.

My understanding is that there's not much of a point in processing the binary further is this check fails. At the same time, as it's not 100% related to this PR, I can drop it if you prefer as well.

[1]: https://github.com/grafana/beyla/pull/458#issuecomment-1823142225